tools/vsextension/ProjectWizard/ImportEnclaveCommand.cs

```diff
@@ -0,0 +1,471 @@
+                    {
+                        var config3 = config as VCConfiguration3;
+                        string name = config.Name;
+
+                        if (name.Contains("ARM"))
+                        {
+                            var clRule = config.Rules.Item("CL") as IVCRulePropertyStorage;
+                            string value = clRule.GetUnevaluatedPropertyValue("PreprocessorDefinitions");
+                            clRule.SetPropertyValue("PreprocessorDefinitions", "_ARM_;" + value);
+
+                            if (isWindows) {
+                                // Enable compiling for ARM.
+                                config3.SetPropertyValue("Configuration", true, "WindowsSDKDesktopARMSupport", "true");
+                                config3.SetPropertyValue("Configuration", true, "WindowsSDKDesktopARM64Support", "true");
+                            }
+                        }
+
+                        if (!isWindows && name.Contains("Debug"))
+                        {
+                            // GCC has no preprocessor define for debug mode, but the generated host file
+                            // expects _DEBUG, so set it here.
+                            var clRule = config.Rules.Item("CL") as IVCRulePropertyStorage;
+                            string value = clRule.GetUnevaluatedPropertyValue("PreprocessorDefinitions");
+                            clRule.SetPropertyValue("PreprocessorDefinitions", "_DEBUG;" + value);
+                        }
+
+                        if (isWindows)
+                        {
+                            // Change OutDir to $(SolutionDir)bin\$(Platform)\$(Configuration)\
+                            // so it's the same as the enclave.
+                            config.OutputDirectory = "$(SolutionDir)bin\\$(Platform)\\$(Configuration)\\";
+                        }
+                        else
+                        {
+                            // Add a post-build event to copy the enclave binary to the existing OutDir.
+                            string cmd = "cp $(RemoteRootDir)/" + baseName + "/bin/$(Platform)/$(Configuration)/" + baseName + ".signed $(RemoteOutDir)" + baseName;
+                            var cbeRule = config.Rules.Item("ConfigurationBuildEvents") as IVCRulePropertyStorage;
+                            cbeRule.SetPropertyValue("RemotePostBuildCommand", cmd);
+                            cbeRule.SetPropertyValue("RemotePostBuildMessage", "Copying enclave binary");
+                        }
+
+                        if (name.Contains("OPTEE") || name.Contains("ARM") || isWindows)
+                        {
+                            // Set the debugger's working directory to where the binary is placed.
+                            VCDebugSettings debugging = config.DebugSettings;
+                            debugging.WorkingDirectory = "$(OutDir)";
+                            continue;
+                        }
```

Comment:
Idem.

Reply:
This should still be used (the ARM/OPTEE parts are for Linux)